dhis-2/dhis-api/src/main/java/org/hisp/dhis/user/UserDetails.java

```diff
@@ -195,6 +197,7 @@ static UserDetails createUserDetails(
 
   boolean isSuper();
 
+  @Override
   String getUid();
```

Comment:
We could only be certain that this is a valid UID if this had the type UID. Is that something we could do later on?

Reply:
I agree.
I am creating the UIDType for Hibernate in another PR, maybe one day we will be able to change the `BaseIdentifiableObject.uid` type to UID.